implementations/rust/ockam/ockam_command/src/node/show.rs

```diff
@@ -52,10 +52,51 @@ async fn run_impl(
     ctx: Context,
     (opts, cmd): (CommandGlobalOpts, ShowCommand),
 ) -> miette::Result<()> {
-    let node_name = get_node_name(&opts.state, &cmd.node_name);
+    if let Some(node_name) = cmd.node_name {
+        show_node(&opts, &ctx, &node_name).await?;
+    } else if !opts.terminal.can_ask_for_user_input() {
+        show_node(&opts, &ctx, &get_default_node_name(&opts.state)).await?;
```

Comment:
These two blocks could be reduced to one, and you can also remove the `else` block to avoid an unnecessary indentation level (which makes the code easier to read).

```rust
if !opts.terminal.can_ask_for_user_input() {
    show_node(&opts, &ctx, &get_node_name(&opts.state, &cmd.node_name)).await?;
    return Ok(());
}

let node_names = opts.state.nodes.list_items_names()?;
...
```